x-pack/plugins/security_solution/common/detection_engine/rule_monitoring/api/get_rule_execution_results/request_schema.test.ts

```diff
@@ -0,0 +1,271 @@
+      });
+
+      describe('when input is a string which is a comma-separated array of invalid values', () => {
+        const cases = [
+          {
+            input: 'value 1,5',
+            expectedErrors: [
+              'Invalid value "value 1" supplied to "DefaultCsvArray<RuleExecutionStatus>"',
+              'Invalid value "5" supplied to "DefaultCsvArray<RuleExecutionStatus>"',
+            ],
+          },
+          {
+            input: 'value 1,succeeded',
+            expectedErrors: [
+              'Invalid value "value 1" supplied to "DefaultCsvArray<RuleExecutionStatus>"',
+            ],
+          },
+          {
+            input: ',5,{}',
+            expectedErrors: [
+              'Invalid value "" supplied to "DefaultCsvArray<RuleExecutionStatus>"',
+              'Invalid value "5" supplied to "DefaultCsvArray<RuleExecutionStatus>"',
+              'Invalid value "{}" supplied to "DefaultCsvArray<RuleExecutionStatus>"',
+            ],
+          },
+        ];
+
+        cases.forEach(({ input, expectedErrors }) => {
+          it(`${input}`, () => {
+            const decoded = DefaultRuleExecutionStatusCsvArray.decode(input);
+            const message = pipe(decoded, foldLeftRight);
+
+            expect(getPaths(left(message.errors))).toEqual(expectedErrors);
+            expect(message.schema).toEqual({});
+          });
+        });
+      });
+    });
+
+    describe('Validation returns default value (an empty array)', () => {
+      describe('when input is', () => {
+        const cases = [{ input: null }, { input: undefined }, { input: '' }, { input: [] }];
+
+        cases.forEach(({ input }) => {
+          it(`${input}`, () => {
+            const decoded = DefaultRuleExecutionStatusCsvArray.decode(input);
+            const message = pipe(decoded, foldLeftRight);
+            const expectedOutput: string[] = [];
+
+            expect(getPaths(left(message.errors))).toEqual([]);
+            expect(message.schema).toEqual(expectedOutput);
+          });
+        });
+      });
+    });
+  });
+
+  describe('DefaultSortField', () => {
+    describe('Validation succeeds', () => {
+      describe('when input is a valid sort field', () => {
+        const cases = [
+          { input: 'timestamp' },
+          { input: 'duration_ms' },
+          { input: 'gap_duration_s' },
+          { input: 'indexing_duration_ms' },
+          { input: 'search_duration_ms' },
+          { input: 'schedule_delay_ms' },
+        ];
+
+        cases.forEach(({ input }) => {
+          it(`${input}`, () => {
+            const decoded = DefaultSortField.decode(input);
+            const message = pipe(decoded, foldLeftRight);
+
+            expect(getPaths(left(message.errors))).toEqual([]);
+            expect(message.schema).toEqual(input);
+          });
+        });
+      });
+    });
+
+    describe('Validation fails', () => {
+      describe('when input is an invalid sort field', () => {
+        const cases = [
```

Comment:
@vitaliidm just prefer how the code looks like when it is done manually 🤷‍♂️ 🙂

When done manually:

- test cases are under the test name which improves readability IMHO
- you can use arbitrary objects in your test cases and have full TS support

When using `test.each`:

- cases are above the test name, makes it a little less readable (can be just my subjective opinion)
- in the Jest version we have in Kibana, objects with clear prop names as test cases are not supported, only arrays; so you case arguments can't be referenced by name which is less readable
- if we had the latest version of Jest, still, referencing case arguments by name would be encoded in the test name, which means we'd loose TS support
- one advantage of using `test.each` is that it serializes arguments to a string better than simple string interpolation does it

When saying `It immediately hints about looped test cases` what do you mean by that?

Reply:
> When saying It immediately hints about looped test cases what do you mean by that?

when I see `test.each` I already expect multiple test cases.

when looping manual, it's not that straightforward: you have only one `it`(one can expect here only one test), but in fact you have to look few lines  up to find cases and where it looped, to understand it's not a single test case

Reply:
Just my 2 cents on `test.each`, I try to avoid that construction whenever possible as it has several considerable downsides.

- It makes tests harder to read. People usually read code from top to down. But `test.each` force you to read code from down to top because you need to read first through the test body to understand what values provided in cases mean.
- `test.each` makes tests harder to debug. It becomes not easy to put a breaking point in the test body or isolate a single failing case.
- `test.each` allows developers to create a large number of test cases without any clear difference between them and without any indication of what matters. Like here https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/usage/detections/rules/update_usage.test.ts#L192-L225

Reply:
> - It makes tests harder to read. People usually read code from top to down. But test.each force you to read code from down to top because you need to read first through the test body to understand what values provided in cases mean.
> - test.each makes tests harder to debug. It becomes not easy to put a breaking point in the test body or isolate a single failing case.
> - test.each allows developers to create a large number of test cases without any clear difference between them and without any indication of what matters. Like here https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/usage/detections/rules/update_usage.test.ts#L192-L225

when looping manually through cases are these still valid concerns?

Reply:
> test.each allows developers to create a large number of test cases without any clear difference between them and without any indication of what matters. Like here https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/usage/detections/rules/update_usage.test.ts#L192-L225

🙈

Reply:
> when looping manually through cases are these still valid concerns?

I would say yes.